### PR TITLE
Fix bug if parquet query contains In decimal column filter

### LINKED ISSUE #68403: Parquet query failed with `In` decimal column filter
After https://github.com/ClickHouse/ClickHouse/pull/68135,  let's execute the following sql:

```sql 
CREATE TABLE file_engine_table (a DECIMAL(9,2)) ENGINE=File(Parquet);
INSERT INTO file_engine_table VALUES (2.00), (4.01), (5.00) , (10.00);
select * from file_engine_table where a in (5.00, 6.00, 8.00, 10.000, 20.00);
```

we get following error message:
``` shell
Code: 170. DB::Exception: Bad get: has Decimal256, requested Decimal32: (in file/uri ./store/769/76981421-d626-4639-b709-fd14c149a5ad//data.Parquet): While executing ParquetBlockInputFormat: While executing File. (BAD_GET) (version 24.9.1.1)
```

### PR DESCRIPTION
**Pull request by baibaichen**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

Fix bug if parquet query contains In decimal column filter after https://github.com/ClickHouse/ClickHouse/pull/68135

Closes https://github.com/ClickHouse/ClickHouse/issues/68403

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp [hunk 1]
```diff
@@ -54,6 +54,18 @@ namespace ErrorCodes
         }                                                              \
     } while (false)
 
+static Field createDecimalField(const Int256 & value, UInt64 precision_value, UInt64 scale)
```
Comment (taiyang-li) [inline src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp:57]: precision_value -> precision is better

File: src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp [hunk 2]
```diff
@@ -54,6 +54,18 @@ namespace ErrorCodes
...
+{
+    if (precision_value <= DecimalUtils::max_precision<Decimal32>)
+        return Field(DecimalField<Decimal32>(Decimal32(static_cast<Int32>(value)), static_cast<UInt32>(scale)));
```
Comment (taiyang-li) [inline src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp:60]: no need to wrap with Field, keep it simple

File: src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp (hunk 1 above)
Comment (taiyang-li) [reply src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp:57]: why precision type and scale type is UInt64 ?

Comment (baibaichen) [reply src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp:57]: just refer to  `createDecimal`
```cpp
template <template <typename> typename DecimalType>
inline DataTypePtr createDecimal(UInt64 precision_value, UInt64 scale_value)
```

File: src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp (hunk 2 above)
Comment (baibaichen) [reply src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp]: fix it.

File: src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp (hunk 1 above)
Comment (baibaichen) [reply src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp]: fix it.

**Comment by alexkats** [comment]
Hi, sorry, but this is a dupllicate of a fix in another PR #68131. Closing this one in favor of the original one

**Pull request closed without merging**